contracts/talent/TalentCommunitySale.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+contract TalentCommunitySale is Ownable, ReentrancyGuard {
+  using Math for uint256;
+
+  IERC20 public paymentToken;
+  address public receivingWallet;
+
+  uint32 public constant TIER1_MAX_BUYS = 300;
+  uint32 public constant TIER2_MAX_BUYS = 500;
+  uint32 public constant TIER3_MAX_BUYS = 1250;
+  uint32 public constant TIER4_MAX_BUYS = 500;
+
+  uint32 public tier1Bought;
+  uint32 public tier2Bought;
+  uint32 public tier3Bought;
+  uint32 public tier4Bought;
```

Comment:
Kind of nice to have since it can be computed, but I would like to have a `totalAmountInvested` variable to read it easily